substrate/frame/support/procedural/src/construct_runtime_v2/parse/pallet.rs

```diff
@@ -0,0 +1,174 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::construct_runtime::parse::{Pallet, PalletPart, PalletPartKeyword, PalletPath};
+use quote::ToTokens;
+use syn::{punctuated::Punctuated, spanned::Spanned, token, Error, Ident, PathArguments};
+
+mod keyword {
+	syn::custom_keyword!(frame);
```

Comment:
personal preference sort of thing here but if you `use syn::custom_keyword;` as the first line in this mod you can type less on each of the lines that follow ;)